### ISSUE TITLE
Fail on unsupported options

### ISSUE THREAD

**Issue by mkenigs** [jordansissel/fpm#1931]
Afaict, `fpm` doesn't support options like `--after-remove` for osxpkg, but it doesn't fail when that option is passed, and the docs don't mention that the option isn't supported. Although one possibility would be to more thoroughly document what's supported, failing on unsupported options would have been more helpful for me

**Comment by jordansissel** [comment]
Good point, and I like this idea :)

**Comment by mkenigs** [comment]
I personally would prefer an error because it's really easy to fix

**Comment by jordansissel** [comment]
Thanks for the quick feedback. I think an error also makes sense.

I did some quick research to see if we could even add `--after-remove` support for osxpkg, and it seems like the macOS pkg files don't have this feature. It's possible I missed something while quickly scanning the Apple docs, though.

From the Apple docs: 

> Limitations of Managed Installs
> 
> The Installer application does not support uninstalling products.

References: https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/SoftwareDistribution4/Install_Operations/Install_Operations.html#//apple_ref/doc/uid/TP40004615-CH14-SW1

**Comment by mkenigs** [comment]
Yeah I never found a way so I just bundled an uninstall script in the package

**Comment by jordansissel** [comment]
> bundled an uninstall script in the package

Slightly related, would you like it if fpm could generate an uninstall script for you? Thinking about it, if fpm generated that uninstall script, then we could have fpm also support `--after-remove` by including it in the uninstall script automatically. What do you think?

**Comment by mkenigs** [comment]
I don't think it would be that helpful, because there's no automatic way to invoke it. Unless I'm missing something, I think I'd still have to tell users to run `/path/to/uninstaller`. Whereas with deb or rpm it's very helpful because the uninstall script will get run by apt or dnf.

**Comment by FWDekker** [comment]
In the context of uninstallers for `.pkg` on macOS, [this answer on StackExchange](https://apple.stackexchange.com/a/439198) mentions that some software vendors also distribute a `<software>-uninstall.pkg` which, when "installed", _remove_ the program. I don't personally use macOS so I haven't seen this in practice, but it might be interesting to have fpm create the uninstaller file as well automatically.
